Latest git fails to build on musl/clang/libc++/linux system

Building on a musl (1.1.24) + clang (9.0.0) + libc++ (9.0.0) Exherbo Linux system, the latest git (b765fb4971db28bac608abc5cd856a9cb94fcfe1) fails to build. I have a hunch this is a difference between glibc and BSD libcs that kakoune currently does build on, but I looked in much detail yet.

```text
root@trotsky ~/kakoune # make -C src debug=no CXX=clang++
make: Entering directory '/root/kakoune/src'
clang++  -O3 -pedantic -std=c++17 -g -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-address -frelaxed-template-template-args -MD -MP -MF .alias_registry.opt.d -c -o .alias_registry.opt.o alias_registry.cc
clang++  -O3 -pedantic -std=c++17 -g -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-address -frelaxed-template-template-args -MD -MP -MF .assert.opt.d -c -o .assert.opt.o assert.cc
In file included from assert.cc:4:
In file included from ./buffer_utils.hh:4:
./buffer.hh:57:29: error: unknown type name 'ssize_t'; did you mean 'size_t'?
    using difference_type = ssize_t;
                            ^~~~~~~
                            size_t
/usr/x86_64-pc-linux-musl/lib/llvm/9/lib/clang/9.0.0/include/stddef.h:46:23: note: 'size_t' declared here
typedef __SIZE_TYPE__ size_t;
                      ^
1 error generated.
make: *** [Makefile:111: .assert.opt.o] Error 1
make: Leaving directory '/root/kakoune/src'
```

--- a/src/buffer.hh
+++ b/src/buffer.hh
@@ -12,6 +12,7 @@
 #include "value.hh"
 #include "vector.hh"
 
+#include <sys/types.h>
 #include <ctime>
 
 namespace Kakoune

